### DISCUSSION TITLE
New file

### DISCUSSION THREAD

**Discussion by noa7777** [gsantner/markor#2393] [Ideas]
Can the program execute in the following order when creating a new file: ① Edit file ② Automatically rename? For example, click the plus sign (. md) and directly enter the editing interface. If there is no content when closing, no content will be saved. If there is content when closing, the first line of non empty text will be read, and some content '# #' will be automatically ignored. If the content is too long, extract the first 40 letters and use the processed 'first line' content as the file name. If the file name is duplicated in the parent directory, add a number or a number plus 1 after the file name to achieve differentiation. Finally, create a file (. md) using the processed new file name.

**Comment by gsantner** [comment]
No. The only place where file movement/renaming is available from is at the file manager (files tab).

Editor is a own tool, that is for editing, not file management.

**Reply by sohang3112** [reply]
The problem is that default Android Files editor isn't showing any option to create file either.

**Reply by sohang3112** [reply]
Besides, every single editor supports creating new file because it's such a basic expectation from an editor. This also should be quite easy to add. Other file operations like rename, delete are optional, but creating file is most basic operation.

**Comment by harshad1** [comment]
See https://github.com/gsantner/markor/issues/95#issuecomment-2041572066

If you have never programmed, chat gpt is your friend.

**Comment by harshad1** [comment]
See this also https://github.com/gsantner/markor/discussions/2537

With a tiny bit of effort you can make a script to do 90% of what you want
